rules/no-useless-length-check.js

```diff
@@ -0,0 +1,148 @@
+'use strict';
+const {methodCallSelector, matches, memberExpressionSelector} = require('./selectors/index.js');
+const isSameReference = require('./utils/is-same-reference.js');
+const {getParenthesizedRange} = require('./utils/parentheses.js');
+
+const messages = {
+	'non-zero': '`Array#some()` returns `false` on empty array, the empty check is not needed.',
+	zero: '`Array#every()` returns `true` on empty array, the non-empty check is not needed.'
```

Comment:
```suggestion
	'non-zero': 'The empty check is useless as `Array#some()` returns `false` for an empty array.',
	zero: 'The non-empty check is useless as `Array#every()` returns `true` for an empty array.'
```

?

Reply:
The message seems not right, I'll double-check the logic.

Reply:
The message was wrong about the length check node, fixed.